index.js

```diff
@@ -1,33 +1,56 @@
 const util = require('util');
+const { parse } = require('querystring');
 const assert = require('assert');
 const Minio = require('minio');
 const mime = require('mime');
 
 Minio.Client.prototype.getObject = util.promisify(Minio.Client.prototype.getObject);
 
-const serve = ({endPoint, port, useSSL, accessKey, secretKey, bucketName}) => {
-    const minioClient = new Minio.Client(Object.assign(
-        {
-            endPoint: '127.0.0.1',
-            port: 9000,
-            useSSL: false,
-            accessKey: 'admin',
-            secretKey: 'password'
-        },
-        {
-            endPoint,
-            port,
-            useSSL,
-            accessKey,
-            secretKey
-        }
-    ));
+const serve = ({
+  endPoint,
+  port,
+  useSSL,
+  accessKey,
+  secretKey,
+  bucketName,
+  region,
+  useFileNameFromQuery
+}) => {
+    const connectionInfos = Object.assign(
+      {
+        endPoint: '127.0.0.1',
+        useSSL: false,
+        accessKey: 'admin',
+        secretKey: 'password'
+      },
+      {
+        endPoint,
+        port,
+        useSSL,
+        accessKey,
+        secretKey
+      }
+    );
+    if (port) {
+      connectionInfos.port = port;
+    }
+    if (region) {
+      connectionInfos.region = region;
+    }
+
+    const minioClient = new Minio.Client(connectionInfos);
 
     return async function serve(ctx, next) {
         if (ctx.request.method === 'GET') {
-            const fileInBucket = ctx.request.url.split('?')[0];
+            const splittedUrl = ctx.request.url.split('?');
+            const fileInBucket = splittedUrl[0];
+            const queryParams = splittedUrl.length > 1 ? parse(splittedUrl[1]) : {};
             let dataStream;
 
+            if (useFileNameFromQuery && queryParams.filename) {
```

Comment:
@Its-Alex **thought** why not remove `useFileNameFromQuery` configuration option and always set `Content-Disposition` header if `filename` query param exists?

Reply:
@stephane-klein If I do that, then this will always download files by default

Reply:
@stephane-klein In fact I want two options:

- One option to always download instead of open in browsers
- Another option to change the fileName of the file

If you want to perform change with these two options working you can.

Reply:
@Its-Alex if `queryParams.filename` is empty, then `ctx.set('Content-Disposition',` is not execute, like before this Merge Request 🤔

Reply:
@stephane-klein In this case `One option to always download instead of open in browsers` will not work

Reply:
What do you think about that:

```
ctx.set('Content-Disposition', `${queryParams.contentDisposition ||  'attachment' }; filename="${queryParams.filename}.${mime.getExtension(mime.getType(fileInBucket))}"`);
```

Reply:
I'm going to implement it.

Reply:
My suggestion in this MR : https://github.com/stephane-klein/koa-s3-proxy/pull/4